File: py/2022/day01.py
from aoc import main
from aoc.parse import paras


def total_calories(elves: list[list[int]]) -> list[int]:
    return [sum(elf) for elf in elves]


if __name__ == "__main__":
    main(
        lambda s: max(total_calories(paras(s, int))),
        lambda s: sum(sorted(total_calories(paras(s, int)))[-3:]),
    )

File: ts/2022/day01.ts
import { main } from 'lib/advent';
import { iter } from 'lib/iter';
import { paragraphs } from 'lib/util';

function parse(s: string): number[] {
  return paragraphs(s).map((lines) => iter(lines).map(Number).sum());
}

main(
  (s) => iter(parse(s)).max(),
  (s) =>
    iter(parse(s).sort((a, b) => b - a))
      .take(3)
      .sum()
);
